### PR TITLE
Add React Native integration

### PR DESCRIPTION
**Pull request by markdalgleish**
In order to better support the React Native integration, I also added a `precomputeNumericValues` function to the `core` package since the existing logic returns strings of CSS values which React Native doesn't support. Not 100% on the naming and it's quite likely we'll want to change it, but wanted to at least get a working implementation in there.

To see the example for yourself, assuming you've [set up your environment for React Native](https://reactnative.dev/docs/environment-setup):

```bash
yarn
yarn preconstruct build
yarn workspace react-native-example start
```

For iOS, run `yarn workspace react-native-example ios`, or if that doesn't work, open `examples/react-native/ios/example.xcworkspace` and click the play button in the toolbar (or hit Cmd+R).

For Android, run `yarn workspace react-native-example android` while you've got an Android emulator open.

The example is a bit heavy so I can see that there's room for debate about whether to include it in the repo or not, but I figured that it's important to have a working example we can reference if we want to make changes to the implementation.

### CODE REVIEW

**Comment by jxom** [comment]
lgtm 420 blaze it

File: packages/react-native/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```tsx
+// Heading.tsx
+import type { ReactNode } from 'react';
+import { Text } from 'react-native';
+import { createTextStyle } from '@capsizecss/react-native';
+
+const headingStyle = {
+  fontFamily: 'Roboto',
+  fontWeight: 'bold',
+  ...createTextStyle({
```
Comment (michaeltaranto) [inline packages/react-native/README.md:25]: Love the simplicity and uniformity of this api with the vanilla integration 😘👌

File: packages/core/src/precomputeNumericValues.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { CapsizeOptions } from './types';
+import { normaliseOptions } from './normaliseOptions';
+
+export function precomputeNumericValues(options: CapsizeOptions) {
```
Comment (michaeltaranto) [inline packages/core/src/precomputeNumericValues.ts:4]: I'd like to talk through this one. I think with a little bike shedding we may be able to have just a single precompute api or something maybe a separate one for relative units. Not certain, but worth a shot.

**Review by michaeltaranto — COMMENTED** [review]
Surprised how simple the integration ends up being! Great stuff.

Not sure about this repo owning the react-native example code though, as it looks like something that would likely rot quickly. The boundary looks simple enough, so maybe stand up a separate repo and link to it from the integration repo?

**Comment by tienphaw** [comment]
I made react-native-capsize a while ago, really glad to see React Native getting official support!

> Not sure about this repo owning the react-native example code though, as it looks like something that would likely rot quickly. The boundary looks simple enough, so maybe stand up a separate repo and link to it from the integration repo?

Maybe we could use [Expo](https://expo.dev/) for the example project instead? That would significantly reduce the size of the project and allow people to run the example without Android Studio or XCode.

**Comment by gunnnnii** [comment]
Hey is there still any interest in getting this merged?

I'd be happy to take a look at it if there is. I could set up an Expo project to replace the example and make adjustments to the way this is set up to get it up to date with the current package version. Do you have any more thoughts on `precomputeNumericValues` api at this point?

I can piggy-back off of this PR or make a new one if that sounds more sensible.

**Comment by michaeltaranto** [comment]
For transparency, I spoke to @markdalgleish a while ago about it offline, and I dont think taking on the maintenance of the react-native setup in the examples would be a good move for the project.

Not ruling out a simpler seam in the future, but for now i'll close this one off.

**Pull request closed without merging**